common/src/main/java/whocraft/tardis_refined/common/GravityUtil.java

```diff
@@ -0,0 +1,85 @@
+package whocraft.tardis_refined.common;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.Options;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.Pose;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.AABB;
+import net.minecraft.world.phys.Vec3;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import whocraft.tardis_refined.common.block.device.AntiGravityBlock;
+
+import java.util.Iterator;
+
+import static net.minecraft.core.BlockPos.betweenClosed;
+
+public class GravityUtil {
+
+
+    public static boolean isInAntiGrav(AABB playerBox, AABB box, Level level) {
+        for (BlockPos pos : betweenClosed(new BlockPos((int) box.maxX, (int) box.maxY, (int) box.maxZ), new BlockPos((int) box.minX, (int) box.minY, (int) box.minZ))) {
+            BlockState blockState = level.getBlockState(pos);
+            if (blockState.getBlock() instanceof AntiGravityBlock) {
+                int space = blockState.getValue(AntiGravityBlock.SPACE);
+                if (space > 0 && level.getBlockState(pos.above()).isAir()) {
+                    if (GravityUtil.createGravityBoxFromLevel(level, pos, space).intersects(playerBox)) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+
+    public static AABB createGravityBoxFromLevel(Level level, BlockPos blockPos, int range) {
+        if (level != null) {
+            return new AABB(blockPos).inflate(range, level.getHeight(), range);
+        }
+        return null;
+    }
+
+
+    public static boolean isInGravityShaft(Player player) {
+        AABB aabb = new AABB(new BlockPos( 999, 101, 19), new BlockPos(1002, 70, 17));
```

Comment:
What is the purpose of this AABB? Seems to be hard coded to the corridor hub.

Reply:
No longer required!